### ISSUE TITLE
How to decode last_error?

### ISSUE THREAD

**Issue by mlafeldt** [geofmureithi/apalis#449]
Hey @geofmureithi,

It looks like Apalis always serializes the original error inside FailedError, e.g.

```
  "last_error": "{\"Err\":\"FailedError: unexpected error: Uncaught PermissionDenied: 'Deno.exit' is not allowed in this context.\\n    at Object.exit (ext:runtime/30_os.js:80:5)\\n    at <user_provided_script>:1:24\"}",
```

How are consumers of that string supposed to deserialize it? I want to access the original error type.

Is there a way to remove the error envelope? I added ErrorHandlingLayer, but it did not change how errors are serialized.

**Comment by mlafeldt** [comment]
I'm using the `chore/v0.6.0` branch.

**Comment by geofmureithi** [comment]
This should be straight forward to resolve.

1. We will rename it to last_result to avoid the confusion.
2. You need to deserialize it as a `Result<Res,Err>` instead of Err, as seen [here](https://docs.rs/serde/latest/serde/de/trait.Deserialize.html#impl-Deserialize%3C'de%3E-for-Result%3CT,+E%3E)

Let me know if 2 works because it should work out of the box.

**Comment by geofmureithi** [comment]
Also, please note that the error type does not expect to be `Serialize`, but rather its converted to string and thats what is stored. I might work on something with the error handling layer to customize this.

**Comment by mlafeldt** [comment]
Oh, I'm already trying to deserialize it as `Result`. Works fine for the `Ok` case, but the `FailedError` envelope gives me trouble in case of `Err`.

**Comment by geofmureithi** [comment]
I think I can work on the error handling layer to allow using crates like https://docs.rs/serde-error/latest/serde_error/

**Comment by geofmureithi** [comment]
But its important to note this message on the crate:
> In some other cases it may make sense to serialize the whole causality chain, but most often it makes most sense to just not serialize errors.

**Comment by mlafeldt** [comment]
Yeah, you're probably right. It looks like the original error isn't serialized, so the inner error is just converted using to_string.
